database.go

```diff
@@ -1,39 +1,135 @@
 package ipld_eth_statedb
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multihash"
+
+	"github.com/VictoriaMetrics/fastcache"
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/core/state"
-	"github.com/jackc/pgx/pgxpool"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/statediff/indexer/ipld"
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+const (
+	// Number of codehash->size associations to keep.
+	codeSizeCacheSize = 100000
+
+	// Cache size granted for caching clean code.
+	codeCacheSize = 64 * 1024 * 1024
 )
 
+var (
+	// not found error
+	errNotFound = errors.New("not found")
+)
+
+// Database interface is a union of the subset of the geth state.Database interface required
+// to support the vm.StateDB implementation as well as methods specific to this Postgres based implementation
 type Database interface {
 	ContractCode(addrHash common.Hash, codeHash common.Hash) ([]byte, error)
 	ContractCodeSize(addrHash common.Hash, codeHash common.Hash) (int, error)
```

Comment:
Could we actually remove the unused address param here, since we don't need to satisfy the `state.Database` interface?

Reply:
Yes great point, will tweak this after merging your PR in